lib/iris/tests/integration/test_netcdf__loadsaveattrs.py

```diff
@@ -70,14 +75,38 @@ def local_attr(request):
     return request.param  # Return the name of the attribute to test.
 
 
+def check_captured_warnings(
```

Comment:
We looked at this link:

[How to capture warnings — pytest documentation](https://docs.pytest.org/en/7.0.x/how-to/capture-warnings.html#additional-use-cases-of-warnings-in-tests)

... which made it clear that PyTest expects users to go off-piste for anything beyond the basics, so that makes me more comfortable with this approach.

As for warning categories, @pp-mo assures me that it would be easy enough to adapt the existing routine should we ever need it. And I'm certainly not saying that now is the right time to solve future problems.